src/backend/distributed/commands/create_citus_local_table.c

```diff
@@ -699,6 +739,51 @@ GetExplicitIndexNameList(Oid relationId)
 }
 
 
+/*
+ * GetExplicitStatsNameList returns a list of statistics names for the given list of.
+ * Statistics Oids.
+ */
+static List *
+GetExplicitStatsCommandList(List *statsOidList, uint64 shardId)
+{
+	PushOverrideEmptySearchPath(CurrentMemoryContext);
+
+	List *statsCommandList = NIL;
+	Oid statsOid;
+	foreach_oid(statsOid, statsOidList)
+	{
+		HeapTuple tup = SearchSysCache1(STATEXTOID, ObjectIdGetDatum(statsOid));
+
+		if (!HeapTupleIsValid(tup))
+		{
+			ereport(WARNING, (errmsg("No stats object found with id: %u", statsOid)));
+			continue;
+		}
+
+		Form_pg_statistic_ext statisticsForm = (Form_pg_statistic_ext) GETSTRUCT(tup);
+
+		char *statsName = statisticsForm->stxname.data;
+		Oid statsSchemaOid = statisticsForm->stxnamespace;
+		char *statsSchema = get_namespace_name(statsSchemaOid);
+
+		char *statsNameWithShardId = pstrdup(statsName);
+		AppendShardIdToName(&statsNameWithShardId, shardId);
+		const char *quotedStatsNameWithShardId = quote_identifier(statsNameWithShardId);
+		char *qualifiedStatsName = quote_qualified_identifier(statsSchema, statsName);
+
+		char *renameShardStatsCommand = GetRenameShardStatsCommand(qualifiedStatsName,
+																   quotedStatsNameWithShardId);
```

Comment:
it seems a bit nicer to me if this function does the quoting (e.g. takes schema, old name and new name as parameters), since the reason we're doing the quoting is to generate the command correctly.